functional/tools/workload/workload.go

```diff
@@ -40,32 +41,34 @@ func (w *Workload) RunDaemon() error {
 	// Create timer for timeout
 	timeoutTimer := time.NewTimer(time.Second * time.Duration(w.timeout))
```

Comment:
I think we could avoid using a timer as a timeout for the stream.Recv(), instead we should use context.WithTimeout() (I'm not absolutely sure about this, but it seems to me like it would make sense).

Reply:
There is a subtle difference here in that the the context timeout would apply to the stream as a whole rather than just a single receive operation... for instance, I think if we set context to 1 minute, the test would time out at 1 minute regardless of whether or not it has received updates... the way I have it currently applies the timeout to a single receive operation, which I _think_ was the original intention when this was written.

In reality, I only updated this code because I couldn't get the tests to pass without doing so :) I am not entirely sure if it will work as intended with the functional test suite... was hoping to check that later on

Reply:
Alright, I see.